cli/fossilize_replay.cpp

```diff
@@ -2866,36 +2893,50 @@ struct ThreadedReplayer : StateCreatorInterface
 					                }
 
 					                sync_worker_memory_context(SHADER_MODULE_MEMORY_CONTEXT);
-					                // The shader module memory context recycles itself.
 
-					                for (auto &parent : parents)
-					                {
-						                if (parent.second.info)
-						                {
-							                resolve_shader_modules(parent.second.info);
-							                assert(!work_item_is_derived(parent.second));
-							                enqueue_pipeline(parent.second.hash, parent.second.info,
-							                                 parent.second.pipeline,
-							                                 parent.second.index + hash_offset + start_index,
-							                                 PARENT_PIPELINE_MEMORY_CONTEXT);
-						                }
-					                }
+					                vector<DerivedInfo> ordered_parents;
+					                for (const auto &parent : parents)
+						                ordered_parents.push_back(parent.second);
 
-					                parents.clear();
+					                sort(begin(ordered_parents), end(ordered_parents), [&](const DerivedInfo &a, const DerivedInfo &b) -> bool {
+						                return a.depth > b.depth;
+					                });
 
 					                // We might be pulling in a parent pipeline from another memory context next iteration,
 					                // so we need to wait for all normal memory contexts.
 					                for (unsigned i = 0; i < NUM_PIPELINE_MEMORY_CONTEXTS; i++)
 						                sync_worker_memory_context(i);
+
+					                uint32_t prev_depth = 0xFFFFFFFF;
+					                for (auto &parent : ordered_parents)
+					                {
+						                if (!parent.info)
+							                continue;
+
+						                if (!derived_work_item_is_satisfied(parent, pipelines))
+							                continue;
+
+						                if (prev_depth != parent.depth) {
```

Comment:
Brace on next line.

Reply:
I moved it because it's needed in ENQUEUE_DERIVED_PIPELINES_OFFSET.